python/metatensor-torch/metatensor/torch/atomistic/openmm_interface.py

```diff
@@ -0,0 +1,161 @@
+                    dtype=torch.int32,
+                ),
+            )
+
+        class MetatensorForce(torch.nn.Module):
+
+            def __init__(
+                self,
+                model: torch.jit._script.RecursiveScriptModule,
+                atomic_numbers: List[int],
+                selected_atoms: Optional[Labels],
+            ) -> None:
+                super(MetatensorForce, self).__init__()
+
+                self.model = model
+                self.register_buffer(
+                    "atomic_numbers", torch.tensor(atomic_numbers, dtype=torch.int32)
+                )
+                self.evaluation_options = ModelEvaluationOptions(
+                    length_unit="nm",
+                    outputs={
+                        "energy": ModelOutput(
+                            quantity="energy",
+                            unit="kJ/mol",
+                            per_atom=False,
+                        ),
+                    },
+                    selected_atoms=selected_atoms,
+                )
+
+            def forward(
+                self, positions: torch.Tensor, cell: Optional[torch.Tensor] = None
+            ) -> torch.Tensor:
+                # move labels if necessary
+                selected_atoms = self.evaluation_options.selected_atoms
+                if selected_atoms is not None:
+                    if selected_atoms.device != positions.device:
+                        self.evaluation_options.selected_atoms = selected_atoms.to(
+                            positions.device
+                        )
+
+                if cell is None:
+                    cell = torch.zeros(
+                        (3, 3), dtype=positions.dtype, device=positions.device
+                    )
+
+                # create System
+                system = System(
+                    types=self.atomic_numbers,
+                    positions=positions,
+                    cell=cell,
+                )
+
+                energy = (
+                    self.model(
+                        [system], self.evaluation_options, check_consistency=True
+                    )["energy"]
+                    .block()
+                    .values.reshape(())
+                )
+                return energy
+
+        metatensor_force = MetatensorForce(
+            model,
+            atomic_numbers,
+            selected_atoms,
+        )
+
+        # torchscript everything
+        module = torch.jit.script(metatensor_force)
+
+        # create the OpenMM force
+        force = openmmtorch.TorchForce(module)
+        isPeriodic = (
+            topology.getPeriodicBoxVectors() is not None
+        ) or system.usesPeriodicBoundaryConditions()
+        force.setUsesPeriodicBoundaryConditions(isPeriodic)
+        force.setForceGroup(forceGroup)
+
+        system.addForce(force)
+
+
+MLPotential.registerImplFactory("metatensor", MetatensorPotentialImplFactory)
```

Comment:
this will fail if we don't have OpenMM installed.

Reply:
Gone